### ISSUE TITLE
[Bug][a11y]: Table examples show no way of declaring a cell as the row header

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3394]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Table

### Expected behavior

It should be possible to declare a column as the rowheader for a virtualized table, and the rowheader should be used to help label the checkbox used to select the row.

### Actual behavior

While it is possible to override the default role for `sp-table-cell` of `gridcell` with [`rowheader`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/rowheader_role), documentation and storybook examples show no way of declaring a cell as the row header for the row.
Being able to declare the rowheader is important for setting the label for the checkbox that selects a row.

Additional concerns: the `role` for `sp-table-cell` should be type safe, `'gridcell' | 'rowheader' | 'cell'`. 

### How can we reproduce this issue?

3. Scroll to '....'
4. Check console
5. See error

**Comment by najikahalsema** [comment]
Tasks:
- [ ] Update the type for role to be typesafe
- [ ] Update documentation to include usage of the types 
- [ ] Ensure `rowheader` labels the checkbox

**Comment by najikahalsema** [comment]
Fixed via #4145
